### PR TITLE
Changed ^ to ** in comments

### PR DESCRIPTION
**Pull request by Sumith1896**
Changed ^ to ** in comments of all source as well as test files, 
so as to maintain uniformity throughout.
The printing of power was previously changed from ^ to ** in https://github.com/sympy/symengine/pull/404.

### CODE REVIEW

File: src/mul.cpp [hunk 1]
```diff
@@ -128,7 +128,7 @@ RCP<const SymEngine::Basic> Mul::from_dict(const RCP<const Number> &coef, map_ba
     }
 }
 
-// Mul (t^exp) to the dict "d"
+// Mul (t**exp) to the dict "d"
```
Comment (Sumith1896) [inline src/mul.cpp]: Is this a valid change? I am not sure about this.

File: src/ntheory.cpp [hunk 2]
```diff
@@ -643,7 +643,7 @@ bool crt(const Ptr<RCP<const Integer>> &R, const std::vector<RCP<const Integer>>
         t = rem[i]->as_mpz() - r;
         if (!mpz_divisible_p (t.get_mpz_t(), g.get_mpz_t()))
             return false;
-        r += m * s * (t / g);           // r += m * (m^-1 mod[i]/g)* (rem[i] - r) / g
+        r += m * s * (t / g);           // r += m * (m**-1 mod[i]/g)* (rem[i] - r) / g
```
Comment (Sumith1896) [inline src/ntheory.cpp]: Have a look at this too! I hope this is fine.

File: src/mul.cpp (hunk 1 above)
Comment (certik) [inline src/mul.cpp]: Yes.

File: src/ntheory.cpp (hunk 2 above)
Comment (certik) [inline src/ntheory.cpp]: Yes, I think that's fine.

**Comment by certik** [comment]
Thanks. I don't think this should change any test results, but let's wait for tests anyway.

**Comment by Sumith1896** [comment]
Agreed. Just felt like doing this clean up while reading the code.

**Comment by certik** [comment]
Much appreciated. If you find more things that need cleanup, feel free to do that.

**Comment by Sumith1896** [comment]
Ready to merge @certik

**Comment by certik** [comment]
Thanks!

**Pull request merged**